src/every_query/tasks.py

```
@@ -101,34 +105,187 @@ def build_task_label_matrix(
+    events_df: pl.DataFrame,
+    base_df: pl.DataFrame,
+    query_codes: list[str],
+) -> pl.DataFrame:
+    """Precompute the minimum positive time delta to each query code for every prediction time.
+
+    Returns a wide DataFrame with columns:
+        subject_id, prediction_time, future_duration, <code_1>, <code_2>, ...
+    where each code column contains the minimum duration until that code's next occurrence
+    strictly after prediction_time (or null if it never occurs).
+
+    Uses join_asof(strategy="forward") per code to avoid the O(n_pred × n_events) cross-join
```

Ruff (0.15.7)

[warning] Docstring contains ambiguous `×` (MULTIPLICATION SIGN). Did you mean `x` (LATIN SMALL LETTER X)?

(RUF002)